src/app/errorReports/newErrorReportForm/NewErrorReportForm.jsx

```diff
@@ -0,0 +1,130 @@
+import { selectIsPostingErrorReport } from '../../../reducers';
+import { selectLocationError, selectLocationData } from '../../../selectors/location';
+import LoadingLabel from '../../../components/form/LoadingLabel';
+import ErrorLabel from '../../../components/form/ErrorLabel';
+import ErrorReportText from './ErrorReportText';
+import ErrorReportServices from './ErrorReportServices';
+
+class NewErrorReportForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      errorReportText: '',
+      errorReportGeneralLocationError: false,
+      errorReportServices: [],
+      isErrorReportFinished: false,
+    };
+
+    this.onErrorReportTextChanged = this.onErrorReportTextChanged.bind(this);
+    this.onErrorReportServicesChanged = this.onErrorReportServicesChanged.bind(this);
+    this.onErrorReportGeneralLocationErrorChanged = this.onErrorReportGeneralLocationErrorChanged.bind(this);
+    this.onErrorReportFinished = this.onErrorReportFinished.bind(this);
+    this.onErrorReportSubmitted = this.onErrorReportSubmitted.bind(this);
+  }
+
+  componentDidMount() {
+    if (!this.props.locationData) {
+      this.props.getLocation(this.props.match.params.locationId);
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.isPostingErrorReport && !this.props.isPostingErrorReport) {
+      this.props.history.push(`/errorreports/${this.props.match.params.locationId}/thanks`);
+    }
+  }
+
+  onErrorReportTextChanged(text) {
+    this.setState({ errorReportText: text });
+  }
+
+  onErrorReportServicesChanged(value) {
+    this.setState((state) => {
+      const errorReportServices = state.errorReportServices.includes(value)
+        ? state.errorReportServices.filter(service => service !== value)
+        : [...state.errorReportServices, value];
+      return {
+        errorReportServices,
+      };
+    });
+  }
+
+  onErrorReportGeneralLocationErrorChanged(value) {
+    this.setState({ errorReportGeneralLocationError: value });
+  }
+
+  onErrorReportFinished() {
+    this.setState({ isErrorReportFinished: true });
+  }
+
+  onErrorReportSubmitted() {
+    console.log('onErrorReportSubmitted triggered.');
+
+    const { errorReportText, errorReportGeneralLocationError, errorReportServices } = this.state;
+
+    this.props.postErrorReport(
+      errorReportText,
+      errorReportGeneralLocationError,
+      errorReportServices,
+    );
+  }
+
+  render() {
+    const { isPostingErrorReport, locationError } = this.props;
+
+    if (locationError) {
+      return <ErrorLabel errorMessage={locationError} />;
+    }
+
+    // Comments component checks for (!locationData || isPostingErrorReport)
+    // Not sure what the !locationData check was doing specifically
+    if (isPostingErrorReport) {
+      return (
+        <LoadingLabel>
+          <span>Adding error report...</span>
+        </LoadingLabel>
+      );
+    }
+
+    if (!this.state.isErrorReportFinished) {
```

Comment:
The fact that the transition between selecting checkboxes and entering text is handled by this component's internal state means that if the user hits "Back" while on the text section, they'll go back to location details, not to the checkboxes.
They'll lose whatever checkboxes they selected, and there's certainly a chance they'll feel frustrated and just never submit the report.

Fixing this, however, would require some changes:
* The checkboxes and the text sections would need to be 2 separate routes under the Router
* The state gathered in the checkboxes section would need to be stored in Redux state
* The text section would need to check if checkboxes state exists, and if not, redirect back to that section (this is in case e.g. the user refreshes the page while on the text section, as Redux state isn't currently persisted across refresh)

This probably isn't critical. It can create a pretty bad user experience, but only if the user actually hits Back after selecting the checkboxes, while likely won't be too common.
If you think you can make these changes pretty easily, I'd say go for it. Otherwise, we can probably do without.

Either way, I would keep it for last.

Reply:
Another option (while not exactly fixing the user hits "Back") would be to include a `Back` button between the `Next`/`Done` button and the `Cancel` button. This would enable the user to at least go back and modify their previous selection, without having to hook up Redux/new routes, etc. (though that _is_ probably the optimal end-goal eventually).

Reply:
Currently the user 'mid-way' check isn't being performed (i.e. a user could navigate directly to the `/text` URL endpoint and continue the workflow from there).
This might be an opportunity to refactor in the future.